test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,209 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv1} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv2} WHERE p=1 ALLOW FILTERING"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv3} WHERE p=1 ALLOW FILTERING"))
+
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=2")]
+            assert allN == sorted([x.c for x in cql.execute(f"SELECT c FROM {mv1} WHERE p=2")])
+
+            for i in range(N):
+                assert [2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c={i}")])
+
+            for i in range(N):
+                assert [v] == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv3} WHERE p=2 AND c={i}")])
+
+# The partition key of the view is strictly contained in the base partition key,
+# so multiple base partitions are combined into one view partition.
+# Test deleting a base partition and verify it is deleted correctly in the view.
+def test_base_partition_deletion_with_smaller_view_partition_key(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p1 int, p2 int, c int, primary key ((p1,p2),c)') as table:
+        insert = cql.prepare(f'INSERT INTO {table} (p1,p2,c) VALUES (?,?,?)')
+        with new_materialized_view(cql, table, '*', 'p1,p2,c', 'p1 is not null and p2 is not null and c is not null') as mv:
+            # Insert into two separate base partitions.
+            # In the view, the rows have the same partition key.
+            cql.execute(insert, [0, 0, 10])
+            cql.execute(insert, [0, 1, 20])
+
+            # Delete one of the partitions
+            cql.execute(f"DELETE FROM {table} WHERE p1=0 AND p2=0")
+
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p1=0 AND p2=0"))
+            assert [20] == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p1=0 AND p2=1")]
+
+            assert [(1,20)] == [(x.p2, x.c) for x in cql.execute(f"SELECT p2, c FROM {mv} WHERE p1=0")]
+
+# Test deleting a large partition when there is a view with the same partition
+# key, and verify that view updates metrics is increased by exactly 1. Deleting
+# a partition in this case is expected to generate one view update for deleting
+# the corresponding view partition by a partition tombstone.
+def test_base_partition_deletion_with_same_view_partition_key_metrics(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key (p,c)') as table:
+        # Insert into two base partitions. We will delete one of them
+        v = 42
+        insert = cql.prepare(f'INSERT INTO {table} (p,c,v) VALUES (?,?,{v})')
+        # Create a view with the same partition key as the base.
+        with new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null') as mv:
+            N = 1000
```

Comment:
Please consider if you really need a large number here. Is anything in our implementation different because you used 1000 and not just 10 (or even 2)?

Reply:
I changed to N=101, it's the minimum that works
it seems the view_updates_pushed counter increases by 1 for every 100 row updates

Reply:
Oh! We indeed have code that collects 100 (`max_rows_for_view_updates`) rows an batches them together in one view update. So if the metric you used counts these "updates" and not individual rows, then indeed you need more than 100 rows to see two separate updates.
Please add a comment to the test explaining that and mentioning `max_rows_for_view_updates` - this test can break if this batching logic ever changes :-(
I wish we had a counter for the actual rows, NOT for "updates", so we don't depend on max_rows_for_view_updates.. Please see if we don't have such a metric (maybe we don't, I don't remember).

Reply:
I added a comment to the test